src/Exception/ExceptionHelper.php

```diff
@@ -90,4 +91,40 @@ public static function getHttpErrorException(
             ),
         };
     }
+
+    /**
+     * Takes an OdataError object and returns a custom exception
+     * that is more specific to the error if it has a numeric error code.
+     * Otherwise, it returns an InvalidResponseException
+     *
+     * @param string $methodName the name of the method that received the OdataError
+     */
+    public static function getExceptionFromOdataError(
+        OdataError $odataError,
+        string $methodName
+    ): BadRequestException|
+    NotFoundException|
+    ForbiddenException|
+    UnauthorizedException|
+    HttpException|
+    ConflictException|
+    TooEarlyException|
+    InternalServerErrorException|
+    InvalidResponseException {
+        $error = $odataError->getError();
+        $errorCode = $error->getCode();
+        $errorMessage = $error->getMessage();
+        if (is_numeric($errorCode)) {
+            $genericHttpException = new HttpException(
+                $errorMessage,
+                (int) $errorCode
+            );
+            return self::getHttpErrorException($genericHttpException);
+        }
+        return new InvalidResponseException(
+            "$methodName returned an OdataError with code '" .
+            $errorCode .
+            "' and message '" . $errorMessage . "'"
+        );
```

Comment:
don't we have to return like above `error message, code and exception instance`
```
            500 => new InternalServerErrorException(
                $message,
                $e->getCode(),
                $e
            ),
```

Reply:
Good idea. I will do that...